common/djangoapps/util/keyword_substitution.py

```diff
@@ -53,30 +46,36 @@ def substitute_keywords(string, user=None, course=None):
     the loop below, and reduces the need for piling up if elif else statements
     when the keyword pool grows.
     """
-    if user is None or course is None:
-        # Cannot proceed without course and user information
-        return string
 
-    for key, func in KEYWORD_FUNCTION_MAP.iteritems():
+    # do this lazily to avoid unneeded database hits
+    KEYWORD_FUNCTION_MAP = {
```

Comment:
I think the issue with moving the map definition here is supporting keyword substitution on the studio side with potentially different context / keywords / mappings, but it might not matter right this moment?